lib/BannerPlugin.js

```diff
@@ -12,6 +12,21 @@ const Template = require("./Template");
 const validateOptions = require("schema-utils");
 const schema = require("../schemas/plugins/BannerPlugin.json");
 
+/**
+ * BannerPlugin Options class.
+ * @typedef {Object|string|Function} BannerOptions
+ * @property {string} banner The banner as string, it will be wrapped in a comment
```

Comment:
Could also accept a function here

Reply:
I've added `Function` to BannerOptions --> `@typedef {......|......|Function}`  .
In your opinion where do I have to add `Function`?

Reply:
Not sure about this, but IMO I would define `BannerOptions` like ` @typedef {Object} BannerOptions` with all its properties and then, I would define the constructor param like this:
` @param {BannerOptions|string|Function} options`, but if I do it like this, I get a bunch of TS errors..

Reply:
`options.banner` ...done!